src/index/coinstatsindex.cpp

```diff
@@ -0,0 +1,309 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <chainparams.h>
+#include <coins.h>
+#include <index/coinstatsindex.h>
+#include <serialize.h>
+#include <txdb.h>
+#include <undo.h>
+#include <validation.h>
+
+constexpr char DB_BLOCK_HASH = 's';
+constexpr char DB_BLOCK_HEIGHT = 't';
+
+namespace {
+
+struct DBVal {
+    uint64_t transaction_output_count;
+    uint64_t bogo_size;
+    CAmount total_amount;
```

Comment:
If I understand the logic, `total_amount` is the net increase of coins due to this block?

I don't want to bloat your stats index, but I'd be interested in adding `total_prevout_spent_amount`, `total_new_outputs_ex_coinbase_amount`, and `coinbase_amount` to unpack `total_amount`.

Reply:
It is the total amount of coins in the UTXO set at that block height. So it's accumulative rather than for this specific block. I added the other values in a draft commit, needs some cleanup but ready to test.